Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerAttachmentsView.swift

```diff
@@ -5,91 +5,163 @@
 import StreamChat
 import UIKit
 
-open class MessageComposerAttachmentsView<ExtraData: ExtraDataTypes>: UIView,
+open class MessageComposerAttachmentsView<ExtraData: ExtraDataTypes>: View,
+    UIConfigProvider,
     UICollectionViewDelegate,
     UICollectionViewDataSource {
+    // MARK: - Underlying types
+    
+    public enum State {
+        case documents([(preview: UIImage, name: String, size: Int64)])
+        case media(previews: [UIImage])
+    }
+    
     // MARK: - Properties
-
-    public let uiConfig: UIConfig<ExtraData>
     
-    var previews: [UIImage] = [] {
+    open var minimumInteritemSpacing: CGFloat = 10
+    open var imagePreviewItemSize: CGSize = .init(width: 100, height: 100)
+    open var documentPreviewItemHeight: CGFloat = 70
+    open var maxNumberOfVisibleDocuments: Int = 3
+    
+    override open var intrinsicContentSize: CGSize {
+        switch state {
+        case .media:
+            let height = imagePreviewItemSize.height + layoutMargins.top + layoutMargins.bottom
+            return .init(
+                width: UIView.noIntrinsicMetric,
+                height: height
+            )
+        case let .documents(items):
+            var height: CGFloat
+            if items.count > maxNumberOfVisibleDocuments {
+                let itemsHeight = documentPreviewItemHeight * CGFloat(maxNumberOfVisibleDocuments)
+                let spacings = minimumInteritemSpacing * CGFloat(maxNumberOfVisibleDocuments)
+                height = itemsHeight + spacings + layoutMargins.top + layoutMargins.bottom
+            } else {
+                height = collectionView.contentSize.height
+            }
+            return .init(
+                width: UIView.noIntrinsicMetric,
+                height: height
+            )
+        }
+    }
+    
+    public var state: State = .media(previews: []) {
         didSet {
-            collectionView.reloadData()
+            updateContentIfNeeded()
         }
     }
     
     var didTapRemoveItemButton: ((Int) -> Void)?
     
-    // MARK: - Subviews
-    
-    public private(set) lazy var flowLayout: UICollectionViewFlowLayout = {
-        let flowLayout = UICollectionViewFlowLayout()
-        flowLayout.itemSize = .init(width: 50, height: 70)
-        flowLayout.sectionInset = .init(top: 20, left: 10, bottom: 10, right: 10)
-        flowLayout.minimumInteritemSpacing = 10
-        flowLayout.scrollDirection = .horizontal
+    open var flowLayout: UICollectionViewFlowLayout {
+        var flowLayout = UICollectionViewFlowLayout()
```

Comment:
nit: You can omit this declaration or just declare empty property `var flowLayout: UICollectionViewFlowLayout!` which will be assigned later, because you cannot escape from the switch. Below you are either creating documentLayout which could be returned instantly or re-creating the flowLayout in the `.media` case.

Reply:
It's to separate views already so it's changed 🙂